doc/source/reference/c-api/types-and-structures.rst

```diff
@@ -119,8 +119,11 @@ PyArray_Type and PyArrayObject
        array. When nd is 0, the array is sometimes called a rank-0
        array. Such arrays have undefined dimensions and strides and
        cannot be accessed. Macro :c:data:`PyArray_NDIM` defined in
-       ``ndarraytypes.h`` points to this data member. :c:data:`NPY_MAXDIMS`
-       is the largest number of dimensions for any array.
+       ``ndarraytypes.h`` points to this data member.
+       Although most operations may be limited in dimensionality, we do not
+       advertise a maximum dimension.  Anyone explicitly relying on one
```

Comment:
Is this a remnant of an earlier version that completely removed `NPY_MAXDIMS` from the public API? Maybe rephrase this to say `NPY_MAXDIMS` is there but not to rely on it because it might change or be removed in the future.